Workflows/Rails upgrades.md

```diff
@@ -0,0 +1,286 @@
+
+1. High priority CVEs for dependencies not related to Rails
+2. High priority Brakeman warnings not related to Rails
+3. Rails upgrades
+4. Ruby upgrades
+5. CVE leftovers
+5. Brakeman leftovers
+
+### Estimations
+Estimating upgrades is tough work due to the Rails gravitational pull. Auxiliary gems are tightly coupled to it and so is our application. We can rely on our gut in some cases, but that can be easily thrown off by a seemingly simple one line changelog notice or an outdated auxiliary dependency that will require major application level changes before we can upgrade it.
+
+Splitting the work into phases might help us visualize the scope of work a bit better, but we should also perform a check of potential challenges **in the estimation phase**.
+
+- assess the [current state of the project](#assessing-the-current-state-of-the-project)
+- prepare a [roadmap](#upgrade-strategy) and estimate each phase individually
+- find exactly which gems [explicitly block the Rails upgrade](#explicit-dependency-rails-support) and estimate their upgrades individually
+- find exactly which gems [implicitly block the Rails upgrade](#implicit-dependency-rails-support) and estimate their upgrades individually
+- check the [Rails changelogs](#2-rails-x-upgrade) ahead of time and triage which breaking changes are likely to affect the estimate (e.g. the introduction of Zeitwerk autoloading might require you to rename modules and classes)
+- check in with the rest of the Rails team and ask very specific questions
+  - has anyone performed **this exact** upgrade?
+  - what was the initial estimation and actual time spent on the upgrade?
+  - was the work deployed in phases?
+  - who performed the testing and to what extent?
+  - was the project similar in size or requirements (big/small, internal/client)?
+  - did we discover any gotchas?
+- do a trial upgrade of the Gemfile if possible and note the count of deprecation warnings or application errors when running specs
+- check the dependency project health, projects which haven't been updated for a long time will likely require more work
+- sync with the project team to determine the amount of manual testing you want to do in each specific phase (e.g. testing yourself, small/complete smoke tests by the QA team). Communicate separate development and QA testing estimates
+- sync with the devops team to provide a Ruby installation estimate if applicable
+
+Projects in bad health will require extra care in the estimation phase. The following factors may severly (e.g. 3x) impact the estimate
+
+- low test coverage
+- low internal application functionality knowledge
+- being more than 1 major Rails or Ruby version behind
+- incomplete or proprietary deployment pipeline
+- poor codebase health
+- lack of functional specification
+- forked dependencies
+- unmaintained dependencies (e.g. paperclip)
```

Comment:
I would maybe add this to the list above as well (related to unmaintaned dependencies)

- check which gems are not maintained and/or not compatible with the next Rails version and may need to be replaced with some other gem